### DISCUSSION TITLE
Include Known 3D Points in Bundle Adjustment

### DISCUSSION THREAD

**Discussion by luigicampanaro** [colmap/colmap#2798] [Q&A]
Hi,

I'm using COLMAP's command line for bundle adjustment on some features I've extracted. 

I would like to know if it’s possible to incorporate **known 3D point** coordinates to constrain the results of the **bundle adjustment**.

Thank you in advance for your help!

**Comment by sarlinpe** [comment]
Yes if you define a custom BA, which works through the Python interface: https://github.com/colmap/colmap/blob/main/pycolmap/custom_bundle_adjustment.py

**Comment by luigicampanaro** [comment]
Thank you for your reply! Do you happen to know of any examples I could refer to?
